### ISSUE TITLE
Turning a bordered float into a split creates a glitchy window

### ISSUE THREAD

**Issue by seandewar** [neovim/neovim#19200]
### Neovim version (nvim -v)

v0.8.0-dev (f71d518)

### Vim (not Nvim) behaves the same?

N/A

### Operating system/version

Fedora Workstation 36

### Terminal name/version

GNOME Terminal v3.44.1

### $TERM environment variable

xterm-256color

### Installation

Source

### How to reproduce the issue

```vim
enew
call setline(1, 'oh noes')
call nvim_open_win(0, 1, #{relative: 'editor', width: 1, height: 1, col: 0, row: 0, border: 'single'})
wincmd L
```

Float border is required.

### Expected behavior

No weirdness. Float is turned into a regular split window.

### Actual behavior

Float turns into a regular split, but there's padding around the contents of the buffer and other rendering oddities.

Bisected to the winbar changes at e1bdb2a258cbe6c5cb981acc6bac82cd9e7706fb.
Maybe something borked in `win_split_ins`?

Also, scrolling may also cause issues, though I can't reproduce them 100% reliably.
Sometimes it causes an assertion failure:
```
tui_flush: Assertion `r.bot <= grid->height && r.right <= grid->width'
```

Sometimes it triggers ASAN:
```
=================================================================
==71888==ERROR: AddressSanitizer: heap-buffer-overflow on address 0x62500003ba78 at pc 0x000000513745 bp 0x7f0624046840 sp 0x7f0624046000
READ of size 4284 at 0x62500003ba78 thread T1
    #0 0x513744 in __asan_memcpy (neovim/build/bin/nvim+0x513744)
    #1 0x1bc2cde in ugrid_scroll neovim/src/nvim/ugrid.c:77:5
    #2 0x1b8da0b in tui_grid_scroll neovim/src/nvim/tui/tui.c:1194:3
    #3 0x1bfdb8a in ui_bridge_grid_scroll_event neovim/build/src/nvim/auto/ui_events_bridge.generated.h:206:3
    #4 0xc6d9a2 in multiqueue_process_events neovim/src/nvim/event/multiqueue.c:153:7
    #5 0xc6779e in loop_poll_events neovim/src/nvim/event/loop.c:68:3
    #6 0x1b98b13 in tui_main neovim/src/nvim/tui/tui.c:506:5
    #7 0x1bfa82d in ui_thread_run neovim/src/nvim/ui_bridge.c:106:3
    #8 0x7f0633830e2c in start_thread (/lib64/libc.so.6+0x8ce2c)
    #9 0x7f06338b661f in __GI___clone3 (/lib64/libc.so.6+0x11261f)

0x62500003ba78 is located 0 bytes to the right of 8568-byte region [0x625000039900,0x62500003ba78)
allocated by thread T1 here:
    #0 0x515987 in __interceptor_calloc (neovim/build/bin/nvim+0x515987)
    #1 0x1297a86 in xcalloc neovim/src/nvim/memory.c:137:15
    #2 0x1bc188f in ugrid_resize neovim/src/nvim/ugrid.c:33:22
    #3 0x1b8997b in tui_grid_resize neovim/src/nvim/tui/tui.c:976:3
    #4 0x1bfb48d in ui_bridge_grid_resize_event neovim/build/src/nvim/auto/ui_events_bridge.generated.h:173:3
    #5 0xc6d9a2 in multiqueue_process_events neovim/src/nvim/event/multiqueue.c:153:7
    #6 0xc6779e in loop_poll_events neovim/src/nvim/event/loop.c:68:3
    #7 0x1b98a74 in tui_main neovim/src/nvim/tui/tui.c:499:12
    #8 0x1bfa82d in ui_thread_run neovim/src/nvim/ui_bridge.c:106:3
    #9 0x7f0633830e2c in start_thread (/lib64/libc.so.6+0x8ce2c)

Thread T1 created by T0 here:
    #0 0x48b2c6 in pthread_create (neovim/build/bin/nvim+0x48b2c6)
    #1 0x1ef8c3e in uv_thread_create_ex neovim/.deps/build/src/libuv/src/unix/thread.c:279:9

SUMMARY: AddressSanitizer: heap-buffer-overflow (neovim/build/bin/nvim+0x513744) in __asan_memcpy
Shadow bytes around the buggy address:
  0x0c4a7ffff6f0: 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
  0x0c4a7ffff700: 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
  0x0c4a7ffff710: 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
  0x0c4a7ffff720: 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
  0x0c4a7ffff730: 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
=>0x0c4a7ffff740: 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00[fa]
  0x0c4a7ffff750: fa fa fa fa fa fa fa fa fa fa fa fa fa fa fa fa
  0x0c4a7ffff760: fa fa fa fa fa fa fa fa fa fa fa fa fa fa fa fa
  0x0c4a7ffff770: fa fa fa fa fa fa fa fa fa fa fa fa fa fa fa fa
  0x0c4a7ffff780: fa fa fa fa fa fa fa fa fa fa fa fa fa fa fa fa
  0x0c4a7ffff790: fa fa fa fa fa fa fa fa fa fa fa fa fa fa fa fa
Shadow byte legend (one shadow byte represents 8 application bytes):
  Addressable:           00
  Partially addressable: 01 02 03 04 05 06 07 
  Heap left redzone:       fa
  Freed heap region:       fd
  Stack left redzone:      f1
  Stack mid redzone:       f2
  Stack right redzone:     f3
  Stack after return:      f5
  Stack use after scope:   f8
  Global redzone:          f9
  Global init order:       f6
  Poisoned by user:        f7
  Container overflow:      fc
  Array cookie:            ac
  Intra object redzone:    bb
  ASan internal:           fe
  Left alloca redzone:     ca
  Right alloca redzone:    cb
==71888==ABORTING
```

**Comment by zeertzjq** [comment]
May be fixed by #18774

**Comment by seandewar** [comment]
Awesome, that seems to fix the issue. Thanks!
